### PR TITLE
Create new component for CU004 `<SelectCheck>`

### PR DESCRIPTION
**Pull request by Andresbl123**
implement the `<SelectCheck />` component which allows you to select, through a checkbox, the options you render

The component for the CU004 is not provided by the inube design-system.

Therefore, we need to create a new multiple option multiple answer input.

The input looks like these.

image

### CODE REVIEW

**Review by novadeveloper14 — APPROVED** [review]

File: src/components/inputs/SelectCheck/styles.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import styled from "styled-components";
+import { inube } from "@inube/design-system";
+
+import { ISelectProps } from ".";
+
+interface IStyledInputContainer {
+  $focused: boolean;
+  $status: ISelectProps["status"];
+  disabled: ISelectProps["disabled"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:9]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  $readonly: ISelectProps["readonly"];
+  onClick: ISelectProps["onClick"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:11]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  compact: {
+    height: "40px",
+  },
+  wide: {
+    height: "48px",
+  },
+};
+
+interface IStyledInput {
+  value: ISelectProps["value"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:24]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  $focused: boolean;
+  $size: ISelectProps["size"];
+  disabled: ISelectProps["disabled"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:27]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  $status: ISelectProps["status"];
+  $fullwidth: ISelectProps["fullwidth"];
+  readOnly?: ISelectProps["readonly"];
+  $required: ISelectProps["required"];
+  onClick?: ISelectProps["onClick"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:32]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+};
+
+interface IStyledInput {
+  value: ISelectProps["value"];
+  $focused: boolean;
+  $size: ISelectProps["size"];
+  disabled: ISelectProps["disabled"];
+  $status: ISelectProps["status"];
+  $fullwidth: ISelectProps["fullwidth"];
+  readOnly?: ISelectProps["readonly"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:30]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  $required: ISelectProps["required"];
+  onClick?: ISelectProps["onClick"];
+  onFocus: ISelectProps["onFocus"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:33]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+  onBlur?: ISelectProps["onBlur"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:34]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+  onChange?: ISelectProps["onChange"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:35]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+  autocomplete?: string;
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:36]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+interface IStyledContainer {
+  disabled: ISelectProps["disabled"];
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:40]: This input is define for the html element, therefore, i would delete it in this interface

File: src/components/inputs/SelectCheck/styles.ts [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+  }};
+
+  opacity: ${({ disabled }) => (disabled ? "0.5" : "none")};
+  cursor: ${({ disabled }) => (disabled ? "not-allowed" : "pointer")};
+`;
+
+export const StyledInput = styled.input<IStyledInput>`
+  outline: none;
+  border-radius: 8px;
+  padding: 0px 12px 0px 16px;
```
Comment (carin91) [inline src/components/inputs/SelectCheck/styles.ts:92]: use the tokens

File: src/components/inputs/SelectCheck/OptionItem/index.tsx [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+export interface IOptionItemProps {
+  id: string;
+  label: string;
+  checked?: boolean;
+  onchange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const OptionItem = (props: IOptionItemProps) => {
+  const { id, label, checked, onchange } = props;
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/OptionItem/index.tsx:13]: resuelto

File: src/components/inputs/SelectCheck/OptionItem/index.tsx [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+  return (
+    <StyledOptionItem>
+      <Stack>
+        <StyledInput
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/OptionItem/index.tsx:18]: Si ya que se debe dar el estilo al elemento, ya que por defecto este viene así : 
image

File: src/components/inputs/SelectCheck/styles.ts (hunk 12 above)
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/styles.ts:92]: resuelto

File: src/components/inputs/SelectCheck/OptionItem/props.ts [hunk 15]
```diff
@@ -0,0 +1,26 @@
+export const parameters = {
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/OptionItem/props.ts]: resuelto

File: src/components/inputs/SelectCheck/styles.ts (hunk 8 above)
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/styles.ts:34]: Se eliminan las propiedades que se permiten sin generar error por tipado,  efectivamente styled-component acepta los atributos por default de un elemento html, pero react maneja algunos atributos de los elementos html con camel case eje: html vanila = `<input onclick="" />`, en  React  = <input onClick="" /> y esto ultimo no es valido para styled-component

File: src/components/inputs/SelectCheck/styles.ts (hunk 9 above)
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/styles.ts:35]: resuelto (Tener en cuenta comentario sobre esto )

File: src/components/inputs/SelectCheck/styles.ts (hunk 10 above)
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/styles.ts:36]: resuelto (Tener en cuenta comentario sobre esto )

File: src/components/inputs/SelectCheck/styles.ts (hunk 11 above)
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/styles.ts:40]: resuelto (Tener en cuenta comentario sobre esto )

File: src/components/inputs/SelectCheck/OptionItem/index.tsx [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+  label: string;
+  checked?: boolean;
+  onchange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const OptionItem = (props: IOptionItemProps) => {
+  const { id, label, checked, onchange } = props;
+
+  return (
+    <StyledOptionItem>
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/OptionItem/index.tsx:16]: resuelto

File: src/components/inputs/SelectCheck/interface.tsx [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+  return "large";
+};
+
+const Message = (
+  props: Pick<ISelectProps, "disabled" | "status"> & { message?: string }
+) => {
+  const { disabled, status, message } = props;
+
+  return status !== "pending" ? (
+    <Stack alignItems="center" gap="4px" margin="s050 s0 s0 s200">
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/interface.tsx:35]: resuelto

File: src/components/inputs/SelectCheck/OptionItem/index.tsx [hunk 18]
```diff
@@ -0,0 +1,30 @@
+import { Label, Stack } from "@inube/design-system";
```
[the remark this answers is not in the document]
Comment (Andresbl123) [reply src/components/inputs/SelectCheck/OptionItem/index.tsx:1]: resuelto

**Pull request merged**